File system operations

tripledoublev opened the pull request:
Adds the following operations:

- #[post("/{repo_id}/upload")]
- #[get("/{repo_id}/list_files")]
- #[post("/{repo_id}/download/{file_name}")]
- #[delete("/{repo_id}/delete_file/{file_name}")]

Closes hyphacoop/openarchive-dweb-backend#49

tripledoublev committed:
Creating a group also names it in the backend

tripledoublev committed:
Add DHTEntity import to groups.rs

tripledoublev committed:
Use cryptokey instead of typedkey

tripledoublev committed:
Remove mut as variable does not need to be mutable

tripledoublev committed:
Fix type mismatch by borrowing key in get_group

tripledoublev committed:
Refactor retrieval to use veilid crypto key instead of typed key

tripledoublev committed:
Refactor repo retrieval in get_repo

tripledoublev committed:
Refactor create_repo to align with refactor

tripledoublev committed:
Remove port on backend initialization

tripledoublev committed:
Merge branch 'create-a-repo_#47' into merged

tripledoublev committed:
fix type conversion errors by handling result and dereferencing box<repo>

tripledoublev committed:
add upload_file endpoint

tripledoublev committed:
add list_files endpoint to retrieve a list of files in a repository

tripledoublev committed:
add download_file endpoint to download a file from a repository

tripledoublev committed:
add delete_file endpoint to delete a file from a repository

tripledoublev committed:
add delete and json as use declarations

tripledoublev committed:
Add delete as a use declaration in server.rs

RangerMauve commented on src/repos.rs:
```diff
@@ -41,18 +59,168 @@ async fn create_repo(
+#[post("/{repo_id}/upload")]
+async fn upload_file(
+    path: web::Path<GroupRepoPath>,
+    body: web::Json<UploadFileRequest>,
```
Is there a way we can put the path into the URL and have the body be `application/octet-stream`? Then we'd need to open the body as a reader and upload to the repo as a stream. I'm worried encoding as JSON will have limits on file size since we'd need to base64 encode/decode stuff.

tripledoublev replied:
- Updated upload_file handler to stream file content via web::Payload
- Removed JSON body handling for file content and replaced it with binary streaming

tripledoublev replied:
@RangerMauve , let me know what you think of this approach and if further changes are needed

tripledoublev committed:
stream file content and switch to query-based filename extraction

- Replaced UploadFileRequest with UploadQuery to extract filename from query string
- Updated upload_file handler to stream file content via web::Payload
- Removed JSON body handling for file content and replaced it with binary streaming

tripledoublev committed:
switch download endpoint to GET and return file data as binary response

- updated the download logic to return file content as an application/octet-stream response instead of a JSON message

tripledoublev committed:
Rename upload and download endpoints to /media

tripledoublev committed:
upload endpoint uses file_name as a path parameter

tripledoublev commented:
Closing this PR in favour of #5
